src/Features/CSharp/Portable/CodeRefactorings/NamedArguments/CSharpAddNamedArgumentsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,175 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Composition;
+using System.Diagnostics;
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.CodeRefactorings;
+using Microsoft.CodeAnalysis.CodeRefactorings.NamedArguments;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static Microsoft.CodeAnalysis.CSharp.SyntaxFactory;
+
+namespace Microsoft.CodeAnalysis.CSharp.CodeRefactorings.NamedArguments
+{
+    [ExportCodeRefactoringProvider(LanguageNames.CSharp, Name = nameof(CSharpAddNamedArgumentsCodeRefactoringProvider)), Shared]
+    internal class CSharpAddNamedArgumentsCodeRefactoringProvider : AbstractAddNamedArgumentsCodeRefactoringProvider<ArgumentSyntax>
+    {
+        protected override IEnumerable<ArgumentSyntax> GetNamedArguments(SeparatedSyntaxList<ArgumentSyntax> arguments, ImmutableArray<IParameterSymbol> parameters)
+        {
+            for (int index = 0; index < arguments.Count; ++index)
+            {
+                var argument = arguments[index];
+                if (argument.NameColon != null)
+                {
+                    yield return argument;
+                }
+                else
+                {
+                    var parameter = parameters[index];
+                    var argumentName = NameColon(parameter.Name);
+                    if (parameter.IsParams && arguments.Count - index > 1)
+                    {
+                        var arrayBuilder = ArrayBuilder<ExpressionSyntax>.GetInstance(arguments.Count - index);
+                        for (; index < arguments.Count; ++index)
+                        {
+                            arrayBuilder.Add(arguments[index].Expression);
+                        }
+
+                        var expressions = arrayBuilder.ToArrayAndFree();
+                        var initializer = InitializerExpression(SyntaxKind.ArrayInitializerExpression, SeparatedList(expressions));
+                        var arrayType = ArrayType(ParseTypeName(parameter.Type.ToDisplayString()));
+                        var arrayCreation = ArrayCreationExpression(arrayType, initializer);
+                        yield return argument.WithNameColon(argumentName).WithExpression(arrayCreation).WithTriviaFrom(argument.Expression);
+                        yield break;
+                    }
+                    else
+                    {
+                        var namedArgument = argument.WithNameColon(argumentName);
+
+                        // Preserve trivia when argument is ref or out
+                        if (argument.RefOrOutKeyword == null)
```

Comment:
Why are we not always preserving trivia?  Also you should pass along any annotations that may be present as well